iree/hal/cuda/graph_command_buffer.c

```diff
@@ -315,22 +315,49 @@ static iree_status_t iree_hal_cuda_graph_command_buffer_push_constants(
                           "need cuda implementation");
 }
 
+// Tie together the binding index and its index in |bindings| array.
+typedef struct {
+  uint32_t index;
+  uint32_t binding;
+} iree_hal_cuda_binding_mapping_t;
+
+// Helper to sort the binding based on their binding index.
+static int compare_binding_index(const void* a, const void* b) {
+  const iree_hal_cuda_binding_mapping_t buffer_a =
+      *(const iree_hal_cuda_binding_mapping_t*)a;
+  const iree_hal_cuda_binding_mapping_t buffer_b =
+      *(const iree_hal_cuda_binding_mapping_t*)b;
+  return buffer_a.binding < buffer_b.binding ? -1 : 1;
+}
+
 static iree_status_t iree_hal_cuda_graph_command_buffer_push_descriptor_set(
     iree_hal_command_buffer_t* base_command_buffer,
     iree_hal_executable_layout_t* executable_layout, uint32_t set,
     iree_host_size_t binding_count,
     const iree_hal_descriptor_set_binding_t* bindings) {
   iree_hal_cuda_graph_command_buffer_t* command_buffer =
       iree_hal_cuda_graph_command_buffer_cast(base_command_buffer);
+  // Convention with the compiler side. We map bindings to kernel argument.
+  // We compact the bindings to get a dense set of arguments and keep them order
+  // based on the binding index.
+  // Sort the binding based on the binding index and map the array index to the
+  // argument index.
+  iree_hal_cuda_binding_mapping_t binding_used[max_binding_count];
```

Comment:
this fails to compile for me on MSVC:
```
[build] D:\dev\projects\iree\iree\hal\cuda\graph_command_buffer.c(350): error C2057: expected constant expression
[build] D:\dev\projects\iree\iree\hal\cuda\graph_command_buffer.c(350): error C2466: cannot allocate an array of constant size 0
[build] D:\dev\projects\iree\iree\hal\cuda\graph_command_buffer.c(350): error C2133: 'binding_used': unknown size
```

Reply:
Oops, ok I'll change it back to a define.

Reply:
https://github.com/google/iree/pull/6130